lib/rouge/lexers/sas.rb

```diff
@@ -0,0 +1,563 @@
+          TRANSLATE TRANSTRN TRANWRD TRIGAMMA TRIM TRIMN TRUNC TSO TYPEOF
+          TZONEID TZONENAME TZONEOFF TZONES2U TZONEU2S UNIFORM UPCASE
+          URLDECODE URLENCODE USS UUIDGEN VAR VARFMT VARINFMT VARLABEL
+          VARLEN VARNAME VARNUM VARRAY VARRAYX VARTYPE VERIFY VFORMAT
+          VFORMATD VFORMATDX VFORMATN VFORMATNX VFORMATW VFORMATWX VFORMATX
+          VINARRAY VINARRAYX VINFORMAT VINFORMATD VINFORMATDX VINFORMATN
+          VINFORMATNX VINFORMATW VINFORMATWX VINFORMATX VLABEL VLABELX
+          VLENGTH VLENGTHX VNAME VNAMEX VTYPE VTYPEX VVALUE VVALUEX WEEK
+          WEEKDAY WHICHC WHICHN WTO YEAR YIELDP YRDIF YYQ ZIPCITY
+          ZIPCITYDISTANCE ZIPFIPS ZIPNAME ZIPNAMEL ZIPSTATE
+         )
+      end
+
+      def self.sas_macro_statements
+        # from SAS 9.4 Macro Language Reference
+        # Chapter 12
+        @sas_macro_statements ||= Set.new %w(
+          %COPY %DISPLAY %GLOBAL %INPUT %LET %MACRO %PUT %SYMDEL %SYSCALL
+          %SYSEXEC %SYSLPUT %SYSMACDELETE %SYSMSTORECLEAR %SYSRPUT %WINDOW
+          %ABORT %DO %TO %UNTIL %WHILE %END %GOTO %IF %THEN %ELSE %LOCAL
+          %RETURN
+          %INCLUDE %LIST %RUN
+        )
+        # Omitted:
+        # %label: Identifies the destination of a %GOTO statement.
+        # %MEND
+      end
+
+      def self.sas_macro_functions
+        # from SAS 9.4 Macro Language Reference
+        # Chapter 12
+
+        @sas_macro_functions ||= Set.new %w(
+          %BQUOTE %NRBQUOTE %EVAL %INDEX %LENGTH %QUOTE %NRQUOTE %SCAN
+          %QSCAN %STR %NRSTR %SUBSTR %QSUBSTR %SUPERQ %SYMEXIST %SYMGLOBL
+          %SYMLOCAL %SYSEVALF %SYSFUNC %QSYSFUNC %SYSGET %SYSMACEXEC
+          %SYSMACEXIST %SYSMEXECDEPTH %SYSMEXECNAME %SYSPROD %UNQUOTE
+          %UPCASE %QUPCASE
+        )
+      end
+
+      def self.sas_auto_macro_vars
+        # from SAS 9.4 Macro Language Reference
+        # Chapter 12
+
+        @sas_auto_macro_vars ||= Set.new %w(
+          &SYSADDRBITS &SYSBUFFR &SYSCC &SYSCHARWIDTH &SYSCMD &SYSDATASTEPPHASE &SYSDATE
+          &SYSDATE9 &SYSDAY &SYSDEVIC &SYSDMG &SYSDSN &SYSENCODING &SYSENDIAN &SYSENV
+          &SYSERR &SYSERRORTEXT &SYSFILRC &SYSHOSTINFOLONG &SYSHOSTNAME &SYSINDEX
+          &SYSINFO &SYSJOBID &SYSLAST &SYSLCKRC &SYSLIBRC &SYSLOGAPPLNAME &SYSMACRONAME
+          &SYSMENV &SYSMSG &SYSNCPU &SYSNOBS &SYSODSESCAPECHAR &SYSODSPATH &SYSPARM
+          &SYSPBUFF &SYSPRINTTOLIST &SYSPRINTTOLOG &SYSPROCESSID &SYSPROCESSMODE
+          &SYSPROCESSNAME &SYSPROCNAME &SYSRC &SYSSCP &SYSSCPL &SYSSITE &SYSSIZEOFLONG
+          &SYSSIZEOFPTR &SYSSIZEOFUNICODE &SYSSTARTID &SYSSTARTNAME &SYSTCPIPHOSTNAME
+          &SYSTIME &SYSTIMEZONE &SYSTIMEZONEIDENT &SYSTIMEZONEOFFSET &SYSUSERID &SYSVER
+          &SYSVLONG &SYSVLONG4 &SYSWARNINGTEXT
+        )
+      end
+
+      def self.proc_keywords 
+        # Create a hash with keywords for common PROCs, keyed by PROC name
+        @proc_keywords = Hash.new
```

Comment:
@tomsutch What this does is to assign a new Hash on every call to `:proc_keywords`. Perhaps you meant to use `||=` here as well..:
```suggestion
        @proc_keywords ||= {}
```

Reply:
Thanks, now changed.